staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -160,14 +161,16 @@ type DeltaFIFO struct {
 	lock sync.RWMutex
 	cond sync.Cond
 
-	// We depend on the property that items in the set are in
-	// the queue and vice versa, and that all Deltas in this
-	// map have at least one Delta.
+	// `items` maps keys to Deltas.
+	// `queue` maintains FIFO order of keys for consumption in Pop().
+	// We maintain the property that keys in the `items` and `queue` are
+	// strictly 1:1 mapping, and that all Deltas in `items` should have
```

Comment:
(a) It might be worth more explicitly emphasizing that a given key appears _at most once_ in the queue.  The code enforces this and relies on this, except as follows.

(b) In DeltaFIFO::queueActionLocked, it is contemplated that `len(newDeltas) == 0` and in this case the code asserts that is OK to remove the key from `f.items` but not `f.queue`.  There are also some other places that read these things and are coded as if a key can be in `f.queue` but not `f.items`.  In fact it is never OK to for a key to appear in `f.queue` but not `f.items`.

Reply:
> keys in the `items` and `queue` are strictly 1:1 mapping
Yeah, since `items` is a map, keys are always unique. So the 1:1 mapping in queue also implies the keys in queue are unique. That could be more explicitly called out as well.